include/private/firebird/common.h

```diff
@@ -98,14 +98,15 @@ double round_for_isc(double value)
 }
 
 template<typename T1>
-void to_isc(void * val, XSQLVAR * var, int x_scale = 0)
+void to_isc(void * val, XSQLVAR * var, short x_scale = 0)
 {
     T1 value = *reinterpret_cast<T1*>(val);
     short scale = var->sqlscale + x_scale;
     short type = var->sqltype & ~1;
     long long divisor = 1, multiplier = 1;
 
-    if ((std::numeric_limits<T1>::is_integer == false) && scale >= 0 &&
+    static bool is_integer = std::numeric_limits<T1>::is_integer;
```

Comment:
This is workaround for MSVC C4127 ( constant value in conditional expression ). If if use const (static or non static ) the warning remains an then it is no difference.

Instead of using temporary static variable we could resolve this by using templates at compile time( this way also runtime if is not necessary):

```c++
template<bool cond> struct cond_to_isc {};
template<> struct cond_to_isc<false> 
{
    static void check(short scale, short type)
    {
        if( scale >= 0 && (type == SQL_SHORT || type == SQL_LONG || type == SQL_INT64) )
            throw soci_error("Can't convert non-integral value to integral column type");
    }
};
template<> struct cond_to_isc<true> 
{ 
    static void check(short scale,short type) { SOCI_UNUSED(scale) SOCI_UNUSED(type) } 
};
// and then call like this - This way compiler will call correct check at compile time without runtime check.
cond_to_isc<std::numeric_limits<T1>::is_integer>::check(scale,type);
```